### ISSUE TITLE
Confusions related E_FERMI and NumOccupied tags in "wt.in" file in wannier_tools

### ISSUE THREAD

**Issue by Majeedqau** [quanshengwu/wannier_tools#47]
I am a fresh user of wannier_tools. I am confused to understand some points for which i need your kind help. My confusions are:
(1) The value of E_FERMI that we need to set in "wt.in" file, should be taken from where? It will be taken form DFT-SCF run or from something else?
(2) The number of wannier occupied bands which are set under the NumOccupied tag, should be the number of bands below the Fermi level in the wannier90 band-structure or something else?
(3) Suppose, if someone interested to calculate Chern number for the band located above the Fermi level, then what should we do in that case? In that case either we need to change the value of Fermi level or only the NumOccupied should increase upto the target band?
I would be highly thankful to you if you favour me in this regard. Thank You and looking forward for your nice response

**Comment by quanshengwu** [comment]
(1) Different software packages have different conventions of whether setting the Fermi level to zero in the tight-binding Hamiltonian wannier90_hr.dat or not. For example, VASP doesn't but Wien2k does.  In order to compatible with different DFT codes, you need to set E_FERMI in the wt.in so that the Fermi level of the energy bands obtained with WannierTools is zero. 
If you use VASP, you can take the Fermi level in the OUTCAR file. If you use Wien2k, please set it to zero. 
By the way, if you are confused with this tag, you can set it to zero and compare the energy band with the DFT one. Then you will know how to set it properly. 
E_Fermi will only shift the energy bands by a constant value. 

(2) NumOccupied tag is a special integer number to specify how many "occupied" bands. You don't have to set it unless you want to study the Wannier charge center (Wilson loop), Berry curvature, Berry phase, and Fermi surface.  Here is an example:
http://www.wanniertools.com/input.html#system
When you set NumOccupied, please forget about the band's order from the DFT energy bands. 
(3) In this case, you need to set NumOccupied.

**Comment by Majeedqau** [comment]
@quanshengwu First of all thank you very much for your kind response and explanation provided. In order to more clear myself, I would like to get your input on the points mentioned below: 
(1) So in simple words,  can we say that NumOccupied is just to count the number of bands lying below the target band in the tight binding bandstructure? By target band, I mean the band in which we are interested in to find its Chern number?   
(2) From your explanation, it seems that NumOccupied doesn't depend on the Fermi level, is it true? In other words, is it a wrong concept to think of,  NumOccupied is just the number of bands below the Fermi level? It can be any number depending on the target band. 
(3) If NumOccupied is not the number of bands below the Fermi level then why we mention Fermi energy in the "wt.in" file? In other words, what is the function of Fermi energy in wannier_tools then? 
(3) In the tight binding bandstructure in the example of Bi2Se3 [http://www.wanniertools.com/input.html#system], you have counted each band twice.  Why? Is it because of SOC or some other reason? 
Once again thank you for your response.

**Comment by quanshengwu** [comment]
@Majeedqau 
(1) Exactly. 
(2) Yes.
(3) Well, some properties are related to the real Fermi level such as the Fermi arc, Fermi surface calculations or other properties related to constant energy contour.
(4) Yes, because of SOC. It's easy to count the number of bands if you use xmgrace software. For example
xmgrace bulkek.dat &

In the end, I created an FAQ page for this question.
https://www.wanniertools.org/the-difference-between-e_fermi-and-numoccupied/

**Comment by Majeedqau** [comment]
@quanshengwu, Dear Sir, I got all your points with full clarity. So thank you very much for it. I have some extra confusions which are mentioned below. Please also comment on these when you get some free time.

(1) I read the wannier_tools manual carefully. There are basically two terminologies are using, one is 
   **Bulk** and other is a **Slab**. I am a little bit confused about it.  So here from **Bulk** you mean 3D 
     materials with no vacuume along all directions while from the **slab**, you mean 2D materials (single 
      layer or having some finite thickness) with a vacuume in one direction (say along z-axis)?  

 (2) Since I am mainly interested to look for the topology (such as Berry curvature and Chern number) of 
      2D materials and their related heterostructures, so for 2D materials Should I need to set **Bulk** or 
      **slab** cards in wannier_tools to get Berry curvature and Chern number?  
      In other words:
     (a) Should I need to set **KPATH_BULK** or **KPATH_SLAB** card for 2D materials? 
     (b) Similarly, Should I  need to set **KPLANE_BULK** or **KPLANE_SLAB** card for 2D materials? 

(3) What value of **KPLANE_BULK** card is best to use for 2D materials with the hexagonal unit cell to 
     get the Berry curvature over the whole B.Z (for instance see the picture below).? 
     ![image](https://user-images.githubusercontent.com/38215807/50819114-1d418000-1364-11e9-8123-a23056bfe10c.png)

(4) For 2D materials with hexagonal strtucture, what value of  SURFACE CARD is best to use for Chern 
      number and Berry curvature calculations? 


Thank You and waiting for your nice response.

**Comment by abir200r8** [comment]
I have a question , i comupte wannier90_hr.dat using QE+Wannier90.In wannier90 i used scdm for projections .and i have set the dis_win_max and dis_froz_max and i have 35 wf but now can i get the projectors for wt.in
I will be very greatfull ,if anyone are able to help me on this
